instrumentation/nethttp/client.go

```diff
@@ -223,16 +225,34 @@ func (t *Transport) doRoundTrip(req *http.Request) (*http.Response, error) {
 // Gets the request payload
 func getRequestPayload(req *http.Request, bufferSize int) string {
 	var rqPayload string
-	if req != nil && req.Body != nil && req.Body != http.NoBody && req.GetBody != nil {
-		rqBody, rqErr := req.GetBody()
+	if req != nil && req.Body != nil && req.Body != http.NoBody {
+		getBody := req.GetBody
+		if getBody == nil {
+			if ptr, err := reflection.GetFieldPointerOf(req.Body, "src"); err == nil {
+				reader := *(*io.Reader)(ptr)
+				if limReader, ok := reader.(*io.LimitedReader); ok {
+					if bufReader, ok := limReader.R.(*bufio.Reader); ok {
+						size := bufReader.Buffered()
+						if size < bufferSize {
+							bufferSize = size
+						}
+						rqBodyBuffer, err := bufReader.Peek(bufferSize)
+						if err == nil {
+							return string(bytes.Runes(rqBodyBuffer))
+						}
+					}
+				}
+			}
+			return ""
```

Comment:
```
			if ptr, err := reflection.GetFieldPointerOf(req.Body, "src"); err == nil {
				reader := *(*io.Reader)(ptr)
				if limReader, ok := reader.(*io.LimitedReader); ok {
					if bufReader, ok := limReader.R.(*bufio.Reader); ok {
						size := bufReader.Buffered()
						if size < bufferSize {
							bufferSize = size
						}
						rqBodyBuffer, err := bufReader.Peek(bufferSize)
						if err == nil {
							return string(bytes.Runes(rqBodyBuffer))
						}
					}
				}
			}
			return ""
```

Maybe extract this logic to other function? And make a call like
```
if getBody == nil {
        return getInternalRequestPayload(req.Body)
}
```

Reply:
I would do that if that code is going to be reused, but right now I don't know what we gain we that.

Reply:
It'd easy to be ready, IMO. I've spent a couple of minutes trying to know where the function returns. 
Additionally, maybe to add a comment explaining when this case is possible (GetBody == null)?

Reply:
done.